spring-integration-core/src/main/java/org/springframework/integration/config/AggregatorFactoryBean.java

```diff
@@ -48,128 +49,267 @@
  *
  */
 public class AggregatorFactoryBean extends AbstractSimpleMessageHandlerFactoryBean<AggregatingMessageHandler>
-				implements ApplicationContextAware, BeanNameAware, ApplicationEventPublisherAware {
+		implements ApplicationContextAware, BeanNameAware, ApplicationEventPublisherAware {
 
-	private final AggregatingMessageHandler aggregator;
+	private Object processorBean;
 
-	public AggregatorFactoryBean(Object processor) {
-		this(processor, null);
+	private String methodName;
+
+	private ApplicationContext applicationContext;
+
+	private String beanName;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private AggregatingMessageHandler aggregator;
+
+	private Boolean expireGroupsUponCompletion;
+
+	private Long sendTimeout;
+
+	private String outputChannelName;
+
+	private AbstractMessageHandlerMetrics metrics;
+
+	private DestinationResolver<MessageChannel> channelResolver;
+
+	private Boolean statsEnabled;
+
+	private Boolean countsEnabled;
+
+	private LockRegistry lockRegistry;
+
+	private MessageGroupStore messageStore;
+
+	private CorrelationStrategy correlationStrategy;
+
+	private ReleaseStrategy releaseStrategy;
+
+	private Expression groupTimeoutExpression;
+
+	private List<Advice> forceReleaseAdviceChain;
+
+	private TaskScheduler taskScheduler;
+
+	private MessageChannel discardChannel;
+
+	private String discardChannelName;
+
+	private Boolean sendPartialResultOnExpiry;
+
+	private Long minimumTimeoutForEmptyGroups;
+
+	private Boolean expireGroupsUponTimeout;
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
 	}
 
-	public AggregatorFactoryBean(Object processor, String methodName) {
-		MessageGroupProcessor outputProcessor;
-		if (processor instanceof MessageGroupProcessor) {
-			outputProcessor = (MessageGroupProcessor) processor;
-		}
-		else {
-			if (methodName == null) {
-				outputProcessor = new MethodInvokingMessageGroupProcessor(processor);
-			}
-			else {
-				outputProcessor = new MethodInvokingMessageGroupProcessor(processor, methodName);
-			}
-		}
-		this.aggregator = new AggregatingMessageHandler(outputProcessor);
+	@Override
+	public void setBeanName(String name) {
+		this.beanName = name;
```

Comment:
This one should be pushed to super, too.
Can you explain your concern do not do that right now?
What's the problem with these options? 
Relying on bug?